src/yunohost/tools.py

```diff
@@ -252,29 +252,41 @@ def tools_postinstall(domain, password, ignore_dyndns=False):
         password -- YunoHost admin password
 
     """
-    dyndns = not ignore_dyndns
+    dyndns_provider = "dyndns.yunohost.org"
 
     # Do some checks at first
     if os.path.isfile('/etc/yunohost/installed'):
         raise MoulinetteError(errno.EPERM,
                               m18n.n('yunohost_already_installed'))
-    if len(domain.split('.')) >= 3 and not ignore_dyndns:
+
+    if not ignore_dyndns:
+        # Check if yunohost dyndns can handle the given domain
+        # (i.e. is it a .nohost.me ? a .noho.st ?)
         try:
-            r = requests.get('https://dyndns.yunohost.org/domains')
-        except requests.ConnectionError:
-            pass
-        else:
-            dyndomains = json.loads(r.text)
-            dyndomain  = '.'.join(domain.split('.')[1:])
-
-            if dyndomain in dyndomains:
-                if requests.get('https://dyndns.yunohost.org/test/%s' % domain).status_code == 200:
-                    dyndns = True
-                else:
-                    raise MoulinetteError(errno.EEXIST,
-                                          m18n.n('dyndns_unavailable'))
-            else:
+            is_nohostme_or_nohost = _dyndns_provides(dyndns_provider, domain)
+        # If an exception is thrown, most likely we don't have internet
+        # connectivity or something. Assume that this domain isn't manageable
+        # and inform the user that we could not contact the dyndns host server.
+        except:
+            logger.warning(m18n.n('dyndns_provider_unreachable',
+                                  provider=dyndns_provider))
+            is_nohostme_or_nohost = False
```

Comment:
So the postinstall will continue but domains with 3 or more parts will not be registered. It's ok if it's not a dyndns domain but if it's a noho.st or nohost.me, there will be a problem.
To fix that the user should do a `yunohost dyndns subscribe --domain DOMAIN.TLD` when the server get a possibility to reach the dyndns_provider.

In this situation, we could add a CRON with something like that:
```
try:
    is_nohostme_or_nohost = _dyndns_provides(dyndns_provider, domain)
    if is_nohostme_or_nohost:
         dyndns_subscribe(dyndns_provider, domain)
    delete_the_cron()
```